runtime/src/chpl-gpu.c

```diff
@@ -34,9 +35,41 @@ bool chpl_gpu_debug = false;
 #include "chplcgfns.h"
 #include "chpl-linefile-support.h"
 #include "chpl-env-gen.h"
+#include "chpl-env.h"
 
 void chpl_gpu_init(void) {
-  chpl_gpu_impl_init();
+  chpl_gpu_impl_init(&chpl_gpu_num_devices);
+
+  // override number of devices if applicable
+  const char* env;
+  int32_t num = -1;
+  if ((env = chpl_env_rt_get("NUM_GPUS_PER_LOCALE", NULL)) != NULL) {
+    if (sscanf(env, "%" SCNi32, &num) != 1) {
+      chpl_error("Cannot parse CHPL_RT_NUM_GPUS_PER_LOCALE environment "
```

Comment:
> I think it's great that your validating this, I don't know if this is the right place to do it or not.

This is heavily inspired by https://github.com/chapel-lang/chapel/blob/main/runtime/src/chpl-tasks.c#L56-L89.

> Is there some common location we use for validating environment variables used by the runtime?

By looking at the code I linked above, my guess is "no".

> Should it be done by one of the chplenv scripts?

I am not sure if that's possible. You can set this env while running your application. I don't think application startup uses those scripts in any meaningful way today.

If anything, this could be something we can ask the `ChplConfig` module to do. I am not sure what other environment variables are there to justify that, though.